lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1313,8 +1324,12 @@ SPIRVValue *LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II,
 
     std::vector<std::pair<Decoration, std::string>> Decorations =
         parseAnnotations(AnnotationString);
-
-    addIntelFPGADecorations(SV, Decorations);
+    if (Decorations.empty()) {
```

Comment:
`DecorateUserSemantic` decoration is handled inside `parseAnnotations` function. So, as long as `AnnotationString` is not empty, `Decorations` will always contain something. What is the point of this `if` then?

Reply:
The only case for `DecorateUserSemantic` that is handled in `parseAnnotations` function is when `AnnotationString` contains arbitrary string in `{}` brackets. 
I can remove support of this case from `parseAnnotations`, so Decorations will be empty if `AnnotationString` contains any random string except the case with `{SOME_STRING}` - this situation causes assertion from StringSwitch in `parseAnnotations` function as it can't found correct match.